### PR TITLE
all: move errors into matrix

### LINKED ISSUE #275: mat64: consider moving Error into matrix
When a cmat128 exists, it will probably want to have pretty much all the same error handling as mat64. To avoid the duplication it's probably worth moving the Error type into matrix.

### PR DESCRIPTION
**Pull request by kortschak**
Closes #275.

@btracey Please take a look.

### CODE REVIEW

**Comment by kortschak** [comment]
matrix needs a package doc. I'm not sure what this should be. Probably a general documentation of the (growing) suite. Some of the comment of mat64 may want to be either copied or moved to matrix.

**Comment by btracey** [comment]
Just a crazy idea, but one I thought worth mentioning before we fix the API. Do we want to change mat64 and cmat128 to just fmat and cmat? It's easier to read and type, and more symmetric.

File: mat64/matrix.go [hunk 1]
```diff
@@ -889,115 +888,13 @@ func Trace(a Matrix) float64 {
...
-var condNorm = lapack.MaxRowSum
-
-// condNormTrans is the norm to compute on A^T to get the same result as computing
-// condNorm on A.
-var condNormTrans = lapack.MaxColumnSum
-
-const stackTraceBufferSize = 1 << 20
+const (
+	// condNorm describes the matrix norm to use for computing the condition number.
+	condNorm = lapack.MaxRowSum
```
Comment (btracey) [inline mat64/matrix.go]: This should be exported in Matrix as well, right? This way someone can write
````
// Will Solve Error?
cond := mat64.Cond(a, matrix.CondNorm)
if cond > matrix.ConditionTolerance {
   // preemptively fix things
}
````

**Comment by btracey** [comment]
LGTM except condNorm.

I would guess the documentation should largely be copied, probably after being separated into mat64 specific and not. Mat64 will likely be the first exposure of many to gonum, so it should keep robust documentation in my opinion, even if that causes some copying.

**Comment by kortschak** [comment]
That was what I was thinking for the documentation. It would be good to be able to structure the mat64 docs so that common documentation could be mechanically updated when changes are made. I'll look into that.

I don't like the idea of dropping the precision from the package names.

File: mat64/matrix.go (hunk 1 above)
Comment (kortschak) [inline mat64/matrix.go]: Done.

**Comment by kortschak** [comment]
Conceivably, we could make ConditionTolerance a var, but I don't think it is a good idea, too much opportunity for client cleverness.

**Pull request merged**